custom/icds_reports/utils/aggregation_helpers/agg_ls_data.py

```diff
@@ -0,0 +1,191 @@
+        yield """
+            UPDATE "{tablename}" agg_ls_report
+            SET vhnd_observed = ut.vhnd_observed
+            FROM (
+                SELECT count(*) as vhnd_observed,
+                location_id as supervisor_id
+                FROM "{ls_vhnd_ucr}"
+                WHERE vhnd_date > %(start_date)s AND vhnd_date < %(end_date)s
+                GROUP BY location_id
+            ) ut
+            WHERE agg_ls_report.supervisor_id = ut.supervisor_id
+        """.format(
+            tablename=self.tablename,
+            ls_vhnd_ucr=self._ucr_tablename(ucr_id=self.ls_vhnd_ucr)
+        ), {
+            "start_date": self.month_start,
+            "end_date": self.next_month_start
+        }
+
+        yield """
+            UPDATE "{tablename}" agg_ls_report
+            SET unique_awc_vists = ut.unique_awc_vists
+            FROM (
+                SELECT count(distinct awc_id) as unique_awc_vists,
+                location_id as supervisor_id
+                FROM "{ls_awc_mgt_ucr}"
+                WHERE submitted_on > %(start_date)s AND  submitted_on< %(end_date)s
+                AND location_entered is not null and location_entered <> ''
+                GROUP BY location_id
+            ) ut
+            WHERE agg_ls_report.supervisor_id = ut.supervisor_id
+        """.format(
+            tablename=self.tablename,
+            ls_awc_mgt_ucr=self._ucr_tablename(ucr_id=self.ls_awc_mgt_ucr)
+        ), {
+            "start_date": self.month_start,
+            "end_date": self.next_month_start
+        }
+
+        yield """
+            UPDATE "{tablename}" agg_ls_report
+            SET beneficiary_vists = ut.beneficiary_vists
+            FROM (
+                SELECT
+                count(*) as beneficiary_vists,
+                location_id as supervisor_id
+                FROM "{ls_home_visit_ucr}"
+                WHERE submitted_on > %(start_date)s AND  submitted_on< %(end_date)s
+                AND visit_type_entered is not null AND visit_type_entered <> ''
+                GROUP BY location_id
+            ) ut
+            WHERE agg_ls_report.supervisor_id = ut.supervisor_id
+        """.format(
+            tablename=self.tablename,
+            ls_home_visit_ucr=self._ucr_tablename(ucr_id=self.ls_home_visit_ucr)
+        ), {
+            "start_date": self.month_start,
+            "end_date": self.next_month_start
+        }
+
+    def indexes(self, aggregation_level):
+        indexes = []
+        agg_locations = ['state_id']
+        if aggregation_level > 1:
+            indexes.append('CREATE INDEX ON "{}" (district_id)'.format(self.tablename))
+            agg_locations.append('district_id')
+        if aggregation_level > 2:
+            indexes.append('CREATE INDEX ON "{}" (block_id)'.format(self.tablename))
+            agg_locations.append('block_id')
+        if aggregation_level > 3:
+            indexes.append('CREATE INDEX ON "{}" (supervisor_id)'.format(self.tablename))
```

Comment:
I think `self.tablename` should be `self._tablename_func(aggregation_level`

Reply:
resolved